darts/utils/data/tabularization.py

```diff
@@ -165,144 +165,1181 @@ def _create_lagged_data(
+
+    Parameters
+    ----------
+    series_or_times
+        The `TimeSeries` and/or `pd.Index` values to compute intersection `bounds` for; any provided `None` values
+        are ignored.
+
+    Returns
+    -------
+    bounds
+        Tuple containing the latest `start_time` and earliest `end time` among all specified `timeseries`, in that
+        order. If no potential overlap exists between the specified series, then `None` is returned instead. Similarly,
+        if no non-`None` `series_or_times` were specified, `None` is returned.
+
+    Raises
+    ------
+    TypeError
+        If the series and/or times in `series_or_times` don't all share the same type of `time_index`
+        (i.e. either all `pd.DatetimeIndex` or `pd.RangeIndex`).
 
-        # concatenate static covariates to features
-        return np.concatenate([features, static_covs], axis=1)
+    """
+    start_times, end_times = [], []
+    for val in series_or_times:
+        if (val is not None) and (len(val) > 0):
+            start_times.append(
+                val.start_time() if isinstance(val, TimeSeries) else val[0]
+            )
+            end_times.append(val.end_time() if isinstance(val, TimeSeries) else val[-1])
+    if not start_times:
+        bounds = None
+    else:
+        times_types = [type(time) for time in start_times]
+        raise_if_not(
+            len(set(times_types)) == 1,
+            (
+                "Specified series and/or times must all "
+                "have the same type of `time_index` "
+                "(i.e. all `pd.RangeIndex` or all `pd.DatetimeIndex`)."
+            ),
+        )
+        # If `start_times` empty, no series were specified -> `bounds = (1, -1)` will
+        # be 'converted' to `None` in next line:
+        bounds = (max(start_times), min(end_times)) if start_times else (1, -1)
+        # Specified timeseries share no overlapping periods.
+        if bounds[1] < bounds[0]:
+            bounds = None
+    return bounds
+
+
+def strided_moving_window(
+    x: np.ndarray,
+    window_len: int,
+    stride: int = 1,
+    axis: int = 0,
+    check_inputs: bool = True,
+) -> np.ndarray:
+    """
+    Extracts moving window views of an `x` array along a specified `axis`, where each window is of length `window_len`
+    and consecutive windows are separated by `stride` indices. The total number of extracted windows equals
+    `num_windows = (x.shape[axis] - window_len)//stride + 1`.
+
+    Notes
+    -----
+    This function is similar to `sliding_window_view` in `np.lib.stride_tricks`, except that:
+        1. `strided_moving_window` allows for consecutive windows to be separated by a specified `stride`,
+        whilst `sliding_window_view` does not.
+        2. `strided_moving_window` can only operate along a single axis, whereas `sliding_window_view` can
+        operate along multiple axes.
+    Additionally, unlike `sliding_window_view`, using `strided_moving_window` doesn't require `numpy >= 1.20.0`.
```

Comment:
Looks really good, although if this there was any reason for going with `sliding_window_view` instead we could perfectly bump our Numpy requirement to `1.20.0`.
In fact, now that I see this, I think we should bump it anyway because we use `sliding_window_view` in `darts.ad` 🙀

Reply:
So the key limitation of `sliding_window_view` is that it doesn't allow for stride arguments. With the being said, an easy work around would be just to call `sliding_window_view` and then extract the desired strides, but I'm not sure how much extra overhead that would add. Thoughts?

Reply:
I think your current way of doing things looks like it's the best because it does allow the `stride`, which could come in handy later on if we choose to make it configurable by users. I just wrote this comment in case the Numpy version was the primary reason for your choice, to say that anyway we can (and should in fact) now require `1.20.0`. I guess the only reason for using `sliding_window_view` would be if it provides substantially better performance, otherwise `strided_moving_window()` looks best.